Assets/Scripts/Treasure.cs

```diff
@@ -3,68 +3,66 @@
 using System.Collections.Generic;
 using UnityEngine;
 using TMPro;
+using UnityEngine.SceneManagement;
+using System.Linq;
 
 public class Treasure : MonoBehaviour
 {
     public bool isClosed = true;
     public Sprite openSprite;
     private List<string> loottable;
+    private GameObject GameRound;
 
     // Start is called before the first frame update
     void Start()
     {
-        // generate loottable
-        loottable = new()
-        {
-            "nothing",
-            "a dagger",
-            "an artifact",
-            "a sword",
-            "a shield",
-            "chainmail",
-            "a wand",
-            "a healing potion",
-            "a robe",
-            "leather armor",
-        };
+        GameRound = GameObject.FindGameObjectsWithTag("GameRound").FirstOrDefault();
```

Comment:
```suggestion
        GameRound = FindObjectOfType<GameRound>();;
```
you could get the script right away with this and wouldnt have to GetComponent later . Line14: `private GameRound GameRound`
Line55: `GameRound.ArtifactsToFind--`
also no need to have `FindGameObjectS` because there is only one